src/System.Memory/src/System/Memory.cs

```diff
@@ -16,6 +16,7 @@ namespace System
     /// Memory represents a contiguous region of arbitrary memory similar to Span.
     /// Unlike Span, it is not a byref-like type.
     /// </summary>
+    [DebuggerTypeProxy(typeof(MemoryDebugView<>))]
```

Comment:
Did you consider adding DebuggerDisplay as well?

Reply:
No, when would DebuggerDisplay be useful? I am not very familiar with it and will look into this: https://docs.microsoft.com/en-us/visualstudio/debugger/using-the-debuggerdisplay-attribute

Reply:
> when would DebuggerDisplay be useful?

It controls what's displayed in the tooltip when you hover over an instance, what shows up in the summary in a watch window before you expand the view to see the results of the type proxy, etc.

Reply:
I see. Should we display the same thing as array, i.e. {T[Memory.Length]}?

Reply:
That sounds reasonable.

Reply:
I will file an issue for adding Debugger Display for Span as well (at least for slow span, which has debugger proxy atm).

Reply:
I will file an issue for adding Debugger Display for Span as well (at least for slow span, which has debugger proxy atm).